tests/percy/snapshotdirector.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @typedef {import('./Photographer.js')} Photographer
+ * @typedef {import('./Camera.js')} Camera
+ */
+
+/**
+ * Responsible for determining which snapshots to take
+ */
+class SnapshotDirector {
+  /**
+   * @param {Object} obj
+   * @param {Photographer} obj.photographer 
+   * @param {Camera} obj.camera
+   */
+  constructor({ photographer, camera }) {
+    this._photographer = photographer;
+    this._camera = camera;
+  }
+
+  async direct() {
+    await this._captureHomepage();
+    await this._captureUniversalSearch();
+    await this._captureVerticalSearch();
+    await this._captureVerticalGridSearch();
+    await this._captureVerticalMapSearch();
+    await this._captureVerticalFullPageMapSearch();
+  }
+
+  async _captureHomepage () {
+    await this._photographer.gotoUniversalPage();
+    await this._camera.snapshot('homepage');
+  }
+  
+  async _captureUniversalSearch () {
+    await this._photographer.gotoUniversalPage({ queryParams: 'query=a' });
+    await this._camera.snapshot('universal-search');
+
+    await this._photographer.gotoUniversalPage({ queryParams: 'query=office+sparce'});
+    await this._camera.snapshot('universal-search-spellcheck');
+  }
+  
+  async _captureVerticalSearch () {
+    await this._photographer.gotoVerticalPage({ vertical: 'events', queryParams: 'query=a'});
+    await this._camera.snapshot('vertical-search');
+
+    await this._photographer.gotoVerticalPage({ vertical: 'events', queryParams: 'query=vrginia'});
+    await this._camera.snapshot('vertical-search-spellcheck');
+  }
+  
+  async _captureVerticalGridSearch () {
+    await this._photographer.gotoVerticalPage({ vertical: 'people', queryParams: 'query=a' });
```

Comment:
nit but I feel like a photographer normally is the one holding a camera? Here it's reads like the director is telling the photographer to drive them around while he uses the camera to take pictures which is a little different than what I first expect

Reply:
maybe snapshot directory should be renamed to Photographer and Photographer to something like PageNavigator or SnapshotNavigator (or maybe something like TourGuide if we want to be more colorful?)

Reply:
I'll update it so that the Director is now the Photographer, and the Photographers are now PageNavigators